storagenode/monitor/monitor.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package monitor
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/common/log"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storagenode/inspector"
+	"storj.io/storj/storagenode/pieces"
+)
+
+var (
+	mon = monkit.Package()
+
+	// Error is the default error class for piecestore monitor errors
+	Error = errs.Class("piecestore monitor")
+)
+
+// Config defines parameters for storage node disk and bandwidth usage monitoring.
+type Config struct {
+	Interval time.Duration `help:"how frequently Kademlia bucket should be refreshed with node stats" default:"1h0m0s"`
+}
+
+// Service which monitors disk usage and updates kademlia network as necessary.
+type Service struct {
+	log                *zap.Logger
+	store              *pieces.Store
+	inspector          *inspector.Endpoint
```

Comment:
Services and endpoints shouldn't depend on other endpoints. This can eventually end up causing issues with authentication. Depend on the `pieces.DB` instead or separate inspector into a endpoint and a service

Reply:
ok